src/features/messages/queries/parse.ts

```diff
@@ -37,6 +44,70 @@ export function parseMessageQueryResult(
   return queryResult(multiProvider, scrapedChains, data, parseMessage);
 }
 
+function getTokenSymbolFromWarpRouteChainAddressMap(
+  chainMetadata: ChainMetadata,
+  address: Address,
+  warpRouteChainAddressMap: WarpRouteChainAddressMap,
+) {
+  const { name } = chainMetadata;
+  if (objKeys(warpRouteChainAddressMap).includes(name)) {
+    const chain = warpRouteChainAddressMap[name];
+    if (objKeys(chain).includes(address)) {
+      return chain[address];
+    }
+  }
+
+  return undefined;
+}
+
+export function parseWarpRouteDetails(
```

Comment:
It maybe made sense to include this here when it was being called by `queryResult` but if it's only used by `WarpTransferDetailCard` then I'd argue it doesn't belong here. The code in this file takes DB data and turns it into the explorer's models. This fn takes the explorer's model and return warp route details. Doesn't feel cohesive with the rest of the code here. WDYT?